include/operations/CompoundOperation.hpp

```diff
@@ -103,6 +103,10 @@ class CompoundOperation final : public Operation {
     return controls.erase(it);
   }
 
+  [[nodiscard]] inline bool isControlled() const override {
+    return !controls.empty();
+  }
+
```

Comment:
```suggestion
```
That's the default implementation in the `Operation` class. So it can be removed here.